src/com/twitter/intellij/pants/execution/PantsMakeBeforeRun.java

```diff
@@ -89,9 +90,12 @@ public static void replaceDefaultMakeWithPantsMake(@NotNull Project project, @No
     if (!(runManager instanceof RunManagerImpl)) {
       return;
     }
-    RunManagerImpl runManagerImpl = (RunManagerImpl) runManager;
     RunConfiguration runConfiguration = settings.getConfiguration();
+    replaceDefaultMakeWithPantsMake(project, runConfiguration);
+  }
+
 
+  public static void replaceDefaultMakeWithPantsMake(@NotNull Project project, @NotNull RunConfiguration runConfiguration) {
```

Comment:
If I understand correctly, this operation noops if we have the IntelliJ compiler.
From the name of the function, I wouldn't expect it to noop in any case, so I'd either remove the conditional [here](https://github.com/pantsbuild/intellij-pants-plugin/pull/447/files#diff-77d87e704dfb9b6b30ffaa3a11f7d6cdR132) (and push it to the callsites), or (less prefereably) change the name to reflect that it may not do anything, leaving a comment doc.

Reply:
Yeah good point. It does a bit more than change the build process to `PantsCompile` and changes the working directory to build root. So it will separate them out and wrap the logic of changing build process to a conditional.

Reply:
Thanks!